### PR TITLE
refactor(Core/Item): Add helpers

### PR DESCRIPTION
**Pull request by Kitzunu**
* ItemTemplate
  * Added: 
    * HasFlag(ItemFlags flag) 
    * HasFlag2(ItemFlags2 flag) 
    * HasFlagCu(ItemFlagsCustom flag)
  * Removed
    * enum ItemFlagsExtra - Use already existing enum ItemFlag2

* Item
  * Added: 
    * IsRefundable() 
    * IsBOPTradable() 
    * IsWrapped()

## Changes Proposed:

This PR proposes changes to:
-  [x] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

## Issues Addressed:

- Probably progresses https://github.com/azerothcore/azerothcore-wotlk/issues/19030 or might even close it. Not tested

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [ ] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Entities/Player/Player.cpp [hunk 1]
```diff
@@ -11722,7 +11722,7 @@ void Player::SendInstanceResetWarning(uint32 mapid, Difficulty difficulty, uint3
 
 void Player::ApplyEquipCooldown(Item* pItem)
 {
-    if (pItem->HasFlag(ITEM_FIELD_FLAGS, ITEM_FLAG_NO_EQUIP_COOLDOWN))
+    if (pItem->GetTemplate()->HasFlag(ITEM_FLAG_NO_EQUIP_COOLDOWN))
```
Comment (Kitzunu) [inline src/server/game/Entities/Player/Player.cpp:11725]: Fixes the issue where ITEM_FLAG_NO_EQUIP_COOLDOWN was never caught as previously it was checking the DB flag in item_field_flags.

**Review by Nyeriah — APPROVED** [review]

**Comment by Grimdhex** [comment]
Now, you should add these helpers to codestyle easly :see_no_evil:

**Comment by Kitzunu** [comment]
Where

**Comment by Grimdhex** [comment]
With my last PR : #19714 

Here the code:

```python
# Codestyle patterns checking for ItemFlags helpers
def itemflags_helpers_check(file: io, file_path: str) -> None:
    global error_handler, results
    file.seek(0)  # Reset file pointer to the beginning
    check_failed = False
    # Parse all the file
    for line_number, line in enumerate(file, start = 1):
        if 'HasFlag(ITEM_FIELD_FLAGS, ITEM_FIELD_FLAG_WRAPPED)' in line:
            print(
                f"Please use IsWrapped() instead HasFlag(ITEM_FIELD_FLAGS, ITEM_FIELD_FLAG_WRAPPED): {file_path} at line {line_number}")
            check_failed = True
	if 'HasFlag(ITEM_FIELD_FLAGS, ITEM_FIELD_FLAG_BOP_TRADEABLE)' in line:
            print(
                f"Please use IsBOPTradable() instead HasFlag(ITEM_FIELD_FLAGS, ITEM_FIELD_FLAG_BOP_TRADEABLE): {file_path} at line {line_number}")
            check_failed = True
	if 'HasFlag(ITEM_FIELD_FLAGS, ITEM_FIELD_FLAG_REFUNDABLE)' in line:
            print(
                f"Please use IsRefundable() instead HasFlag(ITEM_FIELD_FLAGS, ITEM_FIELD_FLAG_REFUNDABLE): {file_path} at line {line_number}")
            check_failed = True
    # Handle the script error and update the result output
    if check_failed:
        error_handler = True
        results["NpcFlagHelpers check"] = "Failed"
``` 
and add the function in parsing_files, by excluding Item.h:
```python
if file_name != 'Item.h':
                            itemflags_helpers_check(file, file_path)
```

**Comment by Nyeriah** [comment]
> /home/runner/work/azerothcore-wotlk/azerothcore-wotlk/modules/mod-transmog/src/Transmogrification.cpp:800:26: fatal error: use of undeclared identifier 'ITEM_FLAGS_EXTRA_HORDE_ONLY'
    if ((proto->Flags2 & ITEM_FLAGS_EXTRA_HORDE_ONLY) && player->GetTeamId() != TEAM_HORDE)

**Comment by Kitzunu** [comment]
I fix, I must've missed it

**Comment by Kitzunu** [comment]
https://github.com/azerothcore/mod-transmog/pull/172

**Comment by Nyeriah** [comment]
> /home/runner/work/azerothcore-wotlk/azerothcore-wotlk/modules/mod-ah-bot/src/AuctionHouseBotConfig.cpp:2507:36: fatal error: use of undeclared identifier 'BIND_WHEN_EQUIPED'; did you mean 'BIND_WHEN_EQUIPPED'?
        if (itr->second.Bonding == BIND_WHEN_EQUIPED && !Bind_When_Equipped)
                                   ^~~~~~~~~~~~~~~~~
                                   BIND_WHEN_EQUIPPED
/home/runner/work/azerothcore-wotlk/azerothcore-wotlk/src/server/game/Entities/Item/ItemTemplate.h:97:5: note: 'BIND_WHEN_EQUIPPED' declared here
    BIND_WHEN_EQUIPPED      
    
 Ouuuh boy, here we go again 🤣

**Comment by heyitsbench** [comment]
> Ouuuh boy, here we go again 🤣

https://github.com/azerothcore/mod-transmog/pull/173
Whoopsie. 😛

**Pull request merged**